webservice/models/webservice_backend.py

```diff
@@ -122,8 +121,10 @@ def _get_adapter_protocol(self):
             protocol += f"+{self.auth_type}-{self.oauth2_flow}"
         return protocol
 
-    @api.depends("auth_type")
-    def _compute_oauth2_flow(self):
+    @api.onchange("auth_type")
+    def _onchange_oauth2_auth_type(self):
```

Comment:
IMO it's a breaking change going from a compute function (that will be triggered anytime auth_type is changed) to an onchange function (that will be triggered only on the UI).

Can't you instead redefine the `compute` slot of the field in the server env module to `False`?

Reply:
I still want this to be managed by server env.

Reply:
I tend to agree

Reply:
I got confused, the model is inheriting from server env in this module, so you can ignore my comment

Reply:
I still think that it would be better to take full control instead.
Can't we control it by improving or overriding `_server_env_transform_field_to_read_from_env` ?

Reply:
To be checked too: https://github.com/OCA/web-api/pull/42